### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by mykola-koval**
[DEMO LINK](https://mykola-koval.github.io/react_clock/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,24 +1,36 @@
...
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+        <p>
+          Current time:
+          {' '}
+          {this.state.isClockVisible && <Clock />}
+        </p>
+        <button type="button" onClick={() => this.changeClockVisibility(true)}>Show clock</button>
+        <button type="button" onClick={() => this.changeClockVisibility(false)}>Hide clock</button>
```
Comment (ArturErmolenko) [inline src/App.tsx:30]: 🔥)

**Review by ArturErmolenko — APPROVED** [review]

File: src/App.tsx (hunk 1 above)
Comment (Elena-brv) [inline src/App.tsx:30]:
```suggestion
        <button 
          type="button" 
          onClick={() => this.changeClockVisibility(true)}
        >
          Show clock
        </button>
        <button
          type="button" 
          onClick={() => this.changeClockVisibility(false)}
        >
          Hide clock
        </button>
```

File: src/components/Clock.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+
+type State = {
+  time: Date,
+};
+
+type Props = {
+};
+
+export class Clock extends React.Component<Props, State> {
```
Comment (Elena-brv) [inline src/components/Clock.tsx:10]:
```suggestion
export class Clock extends React.Component<{}, State> {
```

**Review by Elena-brv — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 3]
```diff
@@ -1,24 +1,47 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './components/Clock';
 
-const App: React.FC = () => {
-  const timerId: NodeJS.Timer = setInterval(() => {
-    const date: Date = new Date();
+type State = {
+  isClockVisible?: boolean,
```
Comment (Elena-brv) [inline src/App.tsx:6]:
```suggestion
  isClockVisible: boolean,
```

**Review by Elena-brv — APPROVED** [review]

**Pull request closed without merging**